packages/cli/src/services/access.service.ts

```diff
@@ -0,0 +1,26 @@
+import { Service } from 'typedi';
+import { SharedWorkflowRepository } from '@/databases/repositories/shared-workflow.repository';
+import { UserRepository } from '@/databases/repositories/user.repository';
+import type { User } from '@/databases/entities/user';
+import type { Workflow } from 'n8n-workflow';
+
+@Service()
+export class AccessService {
+	constructor(
+		private readonly userRepository: UserRepository,
+		private readonly sharedWorkflowRepository: SharedWorkflowRepository,
+	) {}
+
+	/** Whether the user has access to the workflow via project and scope. */
+	async hasAccess(userId: User['id'], workflowId: Workflow['id']) {
```

Comment:
I think this checks specifically read access, right? Maybe this should be called `hasReadAccessToWorkflow`

Reply:
Updated to `hasReadAccess` as workflow is already in the arg and type.